logic/src/commonMain/kotlin/com/wire/kalium/logic/data/user/UserRepository.kt

```diff
@@ -175,7 +175,7 @@ internal class UserDataSource internal constructor(
     override suspend fun fetchUsersIfUnknownByIds(ids: Set<UserId>): Either<CoreFailure, Unit> = wrapStorageRequest {
         val qualifiedIDList = ids.map(idMapper::toDaoModel)
         val knownUsers = userDAO.getUsersByQualifiedIDList(ids.map(idMapper::toDaoModel))
-        qualifiedIDList.filterNot { knownUsers.any { userEntity -> userEntity.id == it } }
+        qualifiedIDList.filterNot { knownUsers.any { userEntity -> userEntity.id == it && !userEntity.name.isNullOrBlank() } }
```

Comment:
So the `name` is now used an indicator that this is an incomplete user which needs to be synchronised from the backend. 

I don't think this so nice. I'd either introduce a dedicated flag for this `needsBeUpdatedFrombackend` or we try up hold the variant that only fully complete users are persisted in the database.

We achieve the later by doing something similar to how the `MemberJoinEventHandler`  works when it persists new members, `persistMembers` fetches unknown users before it inserts the new members. We can do the same in `insertConversationFromEvent`.

I suppose `persistConversations` doesn't already do this because it's used by the slow sync and there we want fetch all users in one batch.